snuba/state/rate_limit.py

```diff
@@ -118,7 +118,14 @@ def rate_limit(
         [("bypass_rate_limit", 0), ("rate_history_sec", 3600)]
     )
 
-    if bypass_rate_limit == 1:
+    if (
+        bypass_rate_limit == 1
+        # XXX: This is probably not what we want to do, since this breaks counts of concurrent
+        # threads, and we may want to have different types of ratelimiting in subscriptions
+        # vs
+        # Bypass rate limit if neither limit is set
+        or (rate_limit_params.concurrent_limit is None and rate_limit_params.per_second_limit is None)
+    ):
         yield None
```

Comment:
Btw does this even work? I'd assume we're going to hit the rest of this function once we exit the context manager.

Reply:
😩 

Nope. Good eye.

```python
>>> import contextlib
>>> @contextlib.contextmanager
... def ctx():
...   yield 1
...   yield 2
... 
>>> with ctx() as a:
...   print('hi')
... 
hi
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "/usr/local/Cellar/python/3.7.2_2/Frameworks/Python.framework/Versions/3.7/lib/python3.7/contextlib.py", line 123, in __exit__
    raise RuntimeError("generator didn't stop")
RuntimeError: generator didn't stop
```